### PR TITLE
Implement authentication for PostgreSQL

### LINKED ISSUE #1694: Enable authentication
* [x] Update `connectionStatus` response.
* [x] Update documentation (new page about auth, supported_commands.md)

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Closes #1694.

## Readiness checklist

* [x] I added some tests. More will be needed in #1568.
* [x] I made spot refactorings.
* [ ] I updated user documentation – in the next PR.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [x] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/pg/msg_explain.go [hunk 1]
```diff
@@ -86,17 +84,8 @@ func (h *Handler) MsgExplain(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
...
-	connInfo := conninfo.GetConnInfo(ctx)
-	if connInfo.PeerAddr != nil {
-		if tcpAddr, ok := connInfo.PeerAddr.(*net.TCPAddr); ok {
-			port = int32(tcpAddr.Port)
-		}
-	}
-
 	serverInfo := must.NotFail(types.NewDocument(
 		"host", hostname,
-		"port", port,
```
Comment (AlekSi) [inline internal/handlers/pg/msg_explain.go:99]: That should be server's port, not client's port. I removed it for now

File: internal/handlers/tigris/tigrisdb/tigrisdb.go [hunk 2]
```diff
@@ -32,12 +32,22 @@ type TigrisDB struct {
 }
 
 // New returns a new TigrisDB.
-func New(cfg *config.Driver, logger *zap.Logger) (*TigrisDB, error) {
-	d, err := driver.NewDriver(context.TODO(), cfg)
+//
+// Passed context is used only by the first checking connection.
+// Canceling it after that function returns does nothing.
```
Comment (rumyantseva) [inline internal/handlers/tigris/tigrisdb/tigrisdb.go:37]: Let's also document what `lazy` is for?

**Review by rumyantseva — DISMISSED** [review]
The code makes sense to me for now.
I tried things locally, seems to be working.

File: internal/util/must/must.go [hunk 3]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package must provides Do helper function.
+// Package must provides helper function that panic on error.
```
[the remark this answers is not in the document]
Comment (rumyantseva) [reply internal/util/must/must.go]: Already fixed :)

File: internal/handlers/tigris/tigrisdb/tigrisdb.go (hunk 2 above)
Comment (AlekSi) [reply internal/handlers/tigris/tigrisdb/tigrisdb.go:37]: Done

**Comment by b1ron** [comment]
All seems to be working. I tried to get `mongo-test` working with TLS but I kept receiving an invalid payload error. However, we probably won't ever need to run it with TLS.

**Review by b1ron — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
I didn't check super attentively, but the PR looks good to me overall, and things seem to be fixed.

**Pull request merged**